### ISSUE TITLE
CLI: Handling non-JS files

### ISSUE THREAD

**Issue by probins** [babel/minify#673]
I notice a small difference between `minify` and `babel` with preset. I'm minifying a directory which includes some files which only contain comments (used for creating docs). `babel` accepts these without problem, whereas `minify` fails with 'Syntax error'. Minifying a comment file will create an empty file which serves little purpose, so it's better not to try minifying these. But the error msg is not very helpful, as it doesn't say in which file the syntax error is occurring. Worth catching these errors and skipping files with no code? Or perhaps just document in the Readme that comment-only files should be excluded from the input file list?

**Comment by probins** [comment]
in fact, this issue seems larger than I thought. `minify` seems to fail if there are other non-js files in the directory as well, whereas `babel` just ignores them. With `babel` you can supply `--ignore` and `--copy-files` options to ignore particular files. `minify` readme says it 'takes in the same cli options as running babel on its own', but this seems to be incorrect, as if I try and use `--ignore`, it fails 'Invalid Options passed'.

**Comment by boopathi** [comment]
> Minifying a comment file will create an empty file which serves little purpose, so it's better not to try minifying these

This shouldn't be detected. The user intended to pass this as an argument. So it will go through the minifier. We can maybe NOT output an empty file. But that will be surprising if the the tools around that are written with an assumption that that file exists also, we cannot help in places where the CLI is used as `cat input.js | minify > out.js`. So, this will not be considered for the CLI. 

> it doesn't say in which file the syntax error is occurring

This can be improved. Can you show me an example of what syntax you're trying to minify?

`babel-minify` leaves any experimental features of the language untouched. The parser though by default is configured to use its default - latest stable ES. If you're trying to minify experimental stage-x feature, then CLI isn't the best place to start. You can use `babel-core` with `babel-preset-minify` and configure the `parserOpts`.

**Comment by probins** [comment]
ok, I've tried some minimal test cases and can't get it to fail, so I may be jumping to conclusions as to what is causing the error. I'll have to dig a bit further to try and find out exactly what is failing and under what circumstances.

**Comment by probins** [comment]
right, I've managed to reproduce the problem. It has nothing to do with comments-only files, but to do with non-JS files, so I've changed the issue title.

I'm minifying a directory with several hundred files in many subdirectories. In amongst the JS files, there are a few WebGL files, like [this one](https://probins.gitlab.io/oles/lib/render/webgl/circlereplay/defaultshader.glsl). `babel` with preset ignores these files and only minifies the JS ones. `minify` doesn't and fails with 'SyntaxError: unknown: Unexpected token, expected ; (6:8)'. If I hack `handleFile()` in `fs.js` to only `minify()` filenames which end in `.js`, then it works. Curiously, I tried a simple test with these files in a flat directory structure, and it ignored the non-JS ones, so is it not recognising they are non-JS if they are in a subdirectory?
